src/AppFactory.php

```diff
@@ -204,6 +204,16 @@ public function newUserFromID( $id ) {
 		return User::newFromId( $id );
 	}
 
+	/**
+	 * @param User $user
+	 *
+	 * @return string[]
+	 */
+	public function getUserRights( User $user ) {
+		$permissionManager = MediaWikiServices::getInstance()->getPermissionManager();
+		return $permissionManager->getUserPermissions( $user );
+	}
```

Comment:
This method is a bit weird. Not really a factory. Why not simply inject a `PermissionManager` instance where it is needed?

Reply:
I implemented property injection, to avoid modifying the constructor (rather than passing argument to the constructor)